src/DynamoCore/Logging/DynamoLogger.cs

```diff
@@ -213,6 +214,41 @@ public DynamoLogger(DebugSettings debugSettings, string logDirectory, Boolean is
             }
         }
 
+        /// <summary>
+        /// Initializes a new instance of <see cref="DynamoLogger"/> class
+        /// with specified debug settings and directory where to write logs
+        /// </summary>
+        /// <param name="debugSettings">Debug settings</param>
+        /// <param name="logDirectory">Directory path where log file will be written</param>
+        /// <param name="isTestMode">Test mode is true or false.</param>
+        /// <param name="isCLIMode">We want to allow logging when CLI mode is true even if we are in test mode.</param>
+        /// <param name="isServiceMode">We want restrict logging in service mode to console only due to lambda limitations.</param>
+        public DynamoLogger(DebugSettings debugSettings, string logDirectory, Boolean isTestMode, Boolean isCLIMode, Boolean isServiceMode)
+            : this(debugSettings, logDirectory, isTestMode)
+        {
+            lock (guardMutex)
+            {
+                this.debugSettings = debugSettings;
+                _isDisposed = false;
+
+                WarningLevel = WarningLevel.Mild;
+                Warning = "";
+
+                notifications = new List<NotificationMessage>();
+
+                testMode = isTestMode;
```

Comment:
we should merge all these flags somehow at some point

Reply:
If you throw in an obsolete message and a TODO Dynamo 3.0 it will be easier to find later when we want to refactor all these signatures throughout the code base.